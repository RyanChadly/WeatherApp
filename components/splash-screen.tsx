import React from "react";
import { StyleSheet, Text, View, ActivityIndicator } from "react-native";
import Icon from "react-native-vector-icons/MaterialCommunityIcons";

export const CustomSplashScreen = () => {
  return (
    <View style={styles.splashscreen}>
      <Text style={styles.text}>Weather App</Text>
      <Icon style={styles.icon} size={50} name="sun-thermometer-outline" />
      <ActivityIndicator style={styles.activity} size="large" />
    </View>
  );
};

const styles = StyleSheet.create({
  splashscreen: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "darkslategrey",
  },
  text: {
    color: "darkgrey",
    fontSize: 40,
    textAlign: "center",
    marginTop: 20,
  },
  activity: {
    marginTop: 100,
  },
  icon: {
    marginTop: 20,
    color: "darkgrey",
  },
});
