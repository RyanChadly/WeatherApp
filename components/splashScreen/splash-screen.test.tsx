import React from "react";
import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import { CustomSplashScreen } from "./splash-screen";

describe("CustomSplashScreen", () => {
  it("should render the splash screen correctly", () => {
    const tree = renderer.create(<CustomSplashScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
