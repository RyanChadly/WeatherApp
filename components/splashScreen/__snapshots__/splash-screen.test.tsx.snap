// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSplashScreen should render the splash screen correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "darkslategrey",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "darkgrey",
        "fontSize": 40,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    Weather App
  </Text>
  <Text />
  <ActivityIndicator
    size="large"
    style={
      {
        "marginTop": 100,
      }
    }
  />
</View>
`;
